app/web_ui/src/lib/stores/rag_progress_store.ts

```diff
@@ -25,45 +26,103 @@ interface RagConfigurationProgressState {
   last_started_rag_config_id: string | null
 }
 
+interface ProjectRagProgressState {
+  projects: Record<string, RagConfigurationProgressState>
+}
+
 export const ragProgressStore = createRagProgressStore()
 
-export const allRagConfigs = derived(ragProgressStore, ($store) => {
-  return sortRagConfigs(Object.values($store.rag_configs), "created_at")
-})
+// derived store that provides the rag state for the currently selected project
+export const currentProjectRagProgressStore = derived(
+  [ragProgressStore, ui_state],
+  ([ragProgress, uiState]) => {
+    const currentProjectId = (uiState as UIState).current_project_id
+    if (!currentProjectId) {
+      return ragProgressStore.getProjectState(null)
+    }
+    return (
+      // return the project state if it exists, otherwise return new default project state
+      ragProgress.projects[currentProjectId] ||
+      ragProgressStore.getProjectState(currentProjectId)
+    )
+  },
+)
 
 // cap concurrent EventSource connections browsers have different limits,
 // we keep the limit low enough to be safe for most browsers
 const run_rag_config_with_throttling = createLimiter(MAX_CONCURRENT_RAG_CONFIGS)
 
 function createRagProgressStore() {
-  const { subscribe, set, update } = writable<RagConfigurationProgressState>({
-    // rag configuration id -> rag config
-    rag_configs: {},
+  const { subscribe, set, update } = writable<ProjectRagProgressState>({
+    projects: {},
+  })
 
-    // rag configuration id -> progress; only applies to rag configurations that are running
-    progress: {},
+  function getDefaultProjectState(): RagConfigurationProgressState {
+    return {
+      // rag configuration id -> rag config
+      rag_configs: {},
 
-    // rag configuration id -> logs
-    logs: {},
+      // rag configuration id -> progress; only applies to rag configurations that are running
+      progress: {},
 
-    // rag configuration id -> status
-    status: {},
+      // rag configuration id -> logs
+      logs: {},
 
-    // rag configuration id -> is_running
-    running_rag_configs: {},
+      // rag configuration id -> status
+      status: {},
 
-    // rag configuration id -> is_archived
-    is_archived: {},
+      // rag configuration id -> is_running
+      running_rag_configs: {},
 
-    // error
-    error: null,
+      // rag configuration id -> is_archived
+      is_archived: {},
 
-    // last started rag config id
-    last_started_rag_config_id: null,
-  })
+      // error
+      error: null,
+
+      // last started rag config id
+      last_started_rag_config_id: null,
+    }
+  }
+
+  function getCurrentProjectId(): string | null {
+    return (get(ui_state) as UIState).current_project_id
+  }
+
+  function getProjectState(
+    project_id: string | null,
+  ): RagConfigurationProgressState {
+    if (!project_id) return getDefaultProjectState()
```

Comment:
raise exception? Seems invalid, and will hide that null is a possible value (which is is if we're allowing null project_ids), which will cause bugs